src/main/java/com/sopt/toss/service/ProductService.java

```diff
@@ -46,7 +48,24 @@ public BrandConDetailDto getBrandCon(long userId, Long productId) {
                 .orElseThrow(() -> new NotFoundException(NOT_FOUND_USER_EXCEPTION, NOT_FOUND_USER_EXCEPTION.getMessage()));
         Product product = productRepository.findById(productId)
                 .orElseThrow(() -> new NotFoundException(NOT_FOUND_PRODUCT_EXCEPTION, NOT_FOUND_PRODUCT_EXCEPTION.getMessage()));
-        Like like = likeRepository.findByUserAndProduct(user, product);
-        return BrandConDetailDto.toDto(product, like.isLike());
+        boolean isLike;
+        Like like = likeRepository.findByUserAndProduct(user, product).orElse(null);
+        if(like ==null) isLike = false;
+        else isLike = like.isLike();
+        return BrandConDetailDto.toDto(product, isLike);
+    }
+
+    @Transactional
+    public void patchBrandConLike(long userId, Long productId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new NotFoundException(NOT_FOUND_USER_EXCEPTION, NOT_FOUND_USER_EXCEPTION.getMessage()));
+        Product product = productRepository.findById(productId)
+                .orElseThrow(() -> new NotFoundException(NOT_FOUND_PRODUCT_EXCEPTION, NOT_FOUND_PRODUCT_EXCEPTION.getMessage()));
+        Like like;
+        like = likeRepository.findByUserAndProduct(user, product).orElse(null);
```

Comment:
```suggestion
        Like like = likeRepository.findByUserAndProduct(user, product).orElse(null);
```
64번째 라인을 없애도 될 것 같습니다!

Reply:
이 부분 로직이랑 isLike default 값을 true/false 중 어떤걸로 둬야할 지 고민하다가 의견 참고해서 바꿔봤는데 괜찮은 지 한 번 봐주세요!

[ 기존 ]
isLike default값 = true
데이터 있는 경우 -> like에 담아 isLike 바꾼 후 save
데이터 없는 경우 -> like에 새로운 객체를 만들어 담은 후 save

[ 변경 ]
isLike default값 = false
`Like like = likeRepository.findByUserAndProduct(user, product).orElseGet(() -> Like.toEntity(user, product));`
like 객체에 바로 데이터 있는 경우와 없는 경우 처리해서 담기
`like.setLike(!like.isLike());`
`likeRepository.save(like);`
이후, 좋아요 상태 변경 후 저장